backend/docker.go

```diff
@@ -390,6 +389,9 @@ func (p *dockerProvider) Start(ctx gocontext.Context, startAttributes *StartAttr
 		ctx, dockerConfig, dockerHostConfig, nil, containerName)
 
 	if err != nil {
+		logger.WithField("err", err).Error("couldn't create container")
+		defer p.checkinCPUSets(ctx, cpuSets)
```

Comment:
what about other error paths below? e.g. `p.client.ContainerStart()` we probably want to checkin there too

Reply:
I believe we also need it in the `ctx.Err()` case.